Allow admins to control the behavior of the Electric Gold Pan via the…

KyleEstein opened the pull request:
… legacy-gold-pan configuration option.

## Description
<!-- Please explain why you are making this pull request. -->
<!-- Start writing below this line -->
As reported in Issue #2324 if a server has cargo networks disabled then the change made to fix Issue #1907 makes Electric Gold Pans practically useless. This change adds a "legacy-gold-pan" option to allow a server admin to revert the behavior of the Electric Gold Pan to what it was before commit 4efc10db1576957547ca18a837a089d269fff770.

## Proposed changes
<!-- Please explain what changes you have made to the code. -->
<!-- Start writing below this line -->
Add a new config option boolean named "legacy-gold-pan". If the option is set then the Electric Gold Pan will behave as it did prior to the update for Issue #1907

## Related Issues (if applicable)
<!-- Please tag any Issues related to your Pull Request -->
<!-- Syntax: "Resolves #000" -->
<!-- Start writing below this line -->

## Checklist
<!-- Here is a little checklist you can follow. -->
<!-- Click on these checkboxes after you created the pull request. -->
<!-- Don't worry, these are not requirements. They only serve as guidance. -->
- [X] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [X] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

WalshyDev commented on src/main/resources/config.yml:
```diff
@@ -13,6 +13,7 @@ options:
   legacy-ore-washer: false
   legacy-dust-washer: false
   legacy-ore-grinder: true
+  legacy-gold-pan: false
```
I hate this "legacy" naming options and I do not think this should be in the config. It's specific to the gold pan so it should be in items.yml

Please give this a better name and make it an ItemSetting

KyleEstein replied:
Can you please provide a better name that would be acceptable? I do not wish to waste time creating a different but equally unacceptable alternative. Unless the legacy options for the ore washer, electric dust washer, and ore grinder are updated as well, Slimefun will end up with different methods to accomplish basically the same thing for different machines.

WalshyDev replied:
Just think of an actually descriptive name. Like what the hell is "legacy" supposed to do? Without being told, you don't know.

The existing options have been there for a long time (literally years). All other options are in items.yml, this is the standard. If you want to revert behaviour for other items you check items.yml.

KyleEstein replied:
> The existing options have been there for a long time (literally years).

Yeah, I know. I created them. So it's my fault they exist (well, not legacy-ore-grinder, not sure who added that one) and if it's not right I'll do what I can to correct it. I noticed there is an option that starts with override-, so maybe something along those lines since we want to override the current behavior of the machine and make it behave like it did in an older version of Slimefun (the "older version" part is what made me choose the "legacy-*" option names in the first place.) But the reason we need the old behavior is because without cargo networks the current behavior makes the machine practically useless, so perhaps it should have a name related to "cargo-net-disabled"? Do either of these cause a good descriptive option name to pop into your head?

KyleEstein replied:
Converted to an ItemSetting named "override-output-limit". But did I break this PR with my rebase? Let me know if I should close this PR and create a fresh one.

KyleEstein commented:
I rebased and now there's 8 commits so I don't know if I've completely hosed this PR and should just create a new one or what. Anyway, I added the ItemSetting "override-output-limit" with a nice description in ElectricGoldPan.java. Let me know if this PR is just broken now and if I need to try again.

svr333 reviewed (commented):
I can see why someone would use this setting, so good for me
Though I dont think the changelog is supposed to be changed

KyleEstein commented:
> I can see why someone would use this setting, so good for me Though I dont think the changelog is supposed to be changed

I made that change to resolve the Merge Conflict that was tagged by TheBusyBot. I don't use git and github often so every time I interact with it it feels like I have to learn it all over again. Any suggestions on what I should do to clean up this PR are appreciated. I'm half expecting a comment like "Yeah, this PR is hopelessly broken. Just close it and and create a new one."

Solverbolt commented:
Not entirely sure where to post such a request, i used the search just to find the general idea. but I noticed that the Electric Pan Machine will not interact or work with anything that can empty its inventory, including everything from the cargo section.  Any way to change that, so that maybe using a hopper or output chest will clear its inventory. Currently having to babysit the thing after putting a stack of gravel in defeats the purpose.

TheBusyBiscuit reviewed (approved):
We can argue about "intuitive naming" here but otherwise it looks fine to me
